float32[2] array